From f6fbb406203cd79624cbe3bbfe2ac2a62b8ecf91 Mon Sep 17 00:00:00 2001
From: Wang Yuanheng <wangyuanheng@wxiat.com>
Date: Fri, 29 Jul 2022 17:18:53 +0800
Subject: [PATCH] sw64: do not set devint_wken for guest and emulator
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f6fbb406203cd79624cbe3bbfe2ac2a62b8ecf91
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5PNB8

--------------------------------

It supports host mode only. Qemu reports "unsupported IPU addr"
when it runs a guest os or emulator system.

Signed-off-by: Wang Yuanheng <wangyuanheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/pci.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/sw_64/kernel/pci.c b/arch/sw_64/kernel/pci.c
index fcc6e0f02a93..6cc872ba9ca5 100644
--- a/arch/sw_64/kernel/pci.c
+++ b/arch/sw_64/kernel/pci.c
@@ -614,7 +614,8 @@ void __init sw64_init_arch(void)
 		cpu_num = sw64_chip->get_cpu_num();
 
 		for (node = 0; node < cpu_num; node++) {
-			set_devint_wken(node);
+			if (is_in_host())
+				set_devint_wken(node);
 			rc_enable = sw64_chip_init->pci_init.get_rc_enable(node);
 			if (rc_enable == 0) {
 				printk("PCIe is disabled on node %ld\n", node);
-- 
2.33.0

